lib/Sema/TypeCheckAttr.cpp

```diff
@@ -4581,6 +4583,42 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
   (void)attr->getParameterIndices();
 }
 
+/// Checks if original candidate and registered derivative match in terms
+/// of static declaration. If the original candidate is a constructor or is
+/// defined as static method then the registered derivative is expected to be static.
+/// Otherwise the registered derivative should be an instance method.
+/// \returns true if mismatch is found, false otherwise.
+static bool checkStaticDeclMismatch(AbstractFunctionDecl *originalCandidate,
+ AbstractFunctionDecl *registred) {
+  return (isa<ConstructorDecl>(originalCandidate) || originalCandidate->isStatic()) !=
+      registred->isStatic();
+}
+
+/// Produces diagnostics for mismatch in static/instance method declaration
+/// between original candidate and registered derivative.
+static void diagnoseStaticDeclMismatch(AbstractFunctionDecl *originalCandidate,
+ FuncDecl *registred) {
```

Comment:
Nit: Please run `git-clang-format` on this patch.

Reply:
I made sure the code is clang-formatted. Fixed in cfd7932a507206c8354eda4405a0ce46c7fbafe3.